In [1]:
import numpy as np
from scipy.stats import binom
import seaborn as sb
import matplotlib.pyplot as plt
from scipy.stats import poisson
import pandas as pd

Задачи 1 и 2
Контрольная работа состоит из пяти вопросов. На каждый вопрос приведено четыре варианта ответа, один из которых правильный. Случайная величина X задаёт число правильных ответов при простом угадывании. Найдите математическое ожидание данной случайной величины. Найдите дисперсию случайной величины X.

Подсказка: постройте закон распределения случайной величины X.

In [2]:
binom.pmf(k = (range(0, 6)), n = 5 , p = 0.25 )

array([0.23730469, 0.39550781, 0.26367187, 0.08789062, 0.01464844,
       0.00097656])

Закон распределения:	
    
    0	        1	         2	        3             4          5
0.23730469	 0.39550781 	0.26367187	 0.08789062    0.01464844   0.00097656


M(X) = np = 5 * 0.25 = 1.25

D(X) = np(1 − p) = 5 * 0.25 * 0.75 = 0.9375

Задача 3
Пользователь получает в среднем 10 писем со спамом на свой почтовый ящик за сутки. 
Найти число N, такое, что с вероятностью 0.95 пользователь получит не более N писем со спамом за текущий день.

In [3]:
poisson.cdf(k=range(10, 16), mu = 10)


array([0.58303975, 0.69677615, 0.79155648, 0.86446442, 0.91654153,
       0.9512596 ])

In [4]:
poisson.cdf(k=15, mu=10)

0.9512595966960213

N = 15

Задача 4
Производятся выстрелы по мишени. Вероятность попадания в мишень при одном выстреле равна 0.01. 
Сколько выстрелов нужно сделать чтобы быть уверенным с вероятностью 0.9, что хотя бы 6 раз будет совершено попадание?

Подсказка: 1) "Вероятность попасть k раз при n выстрелах" - на какое распределение это похоже? 
           2) А если нам нужна вероятность P(X >= k), а не P(X = k)? 3) Здесь предстоит немножко покодить.

In [7]:
poisson.cdf(k=6, mu = 3.89)

0.9004696213959408

In [ ]:
Нужно произвести 389 выстрелов 